apps/next/app/dashboard/page.tsx

```diff
@@ -45,30 +46,47 @@ async function findCreateUserWithClerkId({
   return newUser;
 }
 
+type ParamsForFindCreateUserTeam = {
+  userFirstName: string;
+  userId: string;
+  userClerkId: string;
+  userClerkOrganizationId: string | null | undefined;
+};
 async function findCreateUserTeam({
-  teamName,
+  userFirstName,
   userId,
   userClerkId,
-}: {
-  teamName: string;
-  userId: string;
-  userClerkId: string;
-}): Promise<Team> {
-  const usersPersonalTeam: Team | undefined = await db.query.teams
-    .findMany({
-      where: (team, { eq, and }) =>
-        and(eq(team.name, teamName), eq(team.creatorId, userId)),
-    })
-    .then((rows) => rows[0] || undefined);
-  if (usersPersonalTeam) {
-    return usersPersonalTeam;
+  userClerkOrganizationId,
+}: ParamsForFindCreateUserTeam): Promise<Team> {
+  /*
+  - If a user is already has a Metal account and then uses Clerk's <OrganizationSwitcher />
+  to create a new org, the Clerk org will exist but won't have an associated Metal team. This
+  function will handle that.
+  - If a user is newly registered, they won't have any organizations or teams. This
+  function will handle that too.
+   */
+  if (userClerkOrganizationId) {
+    const usersPersonalTeam: Team | undefined = await db.query.teams
+      .findMany({
+        where: (team, { eq, and }) =>
+          and(
+            eq(team.clerkId, userClerkOrganizationId),
+            eq(team.creatorId, userId)
+          ),
+      })
+      .then((rows) => rows[0] || undefined);
+    if (usersPersonalTeam) {
+      return usersPersonalTeam;
+    }
```

Comment:
similarly don't like the extra statefulness of the userClerkOrganizationId param here...

My understanding of the previous version of this function was that it roughly did:
1) try to find a Team that the user created and is part of in the DB. If found, we're done
2) create a new org in Clerk and a new Team in our db

Now there's a new userClerkOrganizationId param that is unclear to me what it means / how it's useful in the context of this function's original purpose of creating a personal team for the user.

I think it'd be clearer to separate this out entirely and maybe create a separate function call that checks that the active clerk org has a corresponding Team in our db, e.g.

```ts
  const userPersonalTeam = await findCreateUserTeam({
    teamName: `${user.firstName}'s Projects`,
    userId: user.id,
    userClerkId: clerkUser.id,
  });

  // new
  const { orgId } = auth();
  if (orgId) {
    await ensureTeamForClerkOrg({ clerkOrgId: orgId })
  }
```

Reply:
Yes I've created `ensureTeamForClerkOrg`, but I still used it from within  `findCreateUserTeam`. This is primarily because I think `ensureTeamForClerkOrg` satisfies the "find" in "findCreateUserTeam". I don't think it's good to have both `ensureTeamForClerkOrg` and `findCreateUserTeam` both contain logic for check-user-team/check-or-create-user-team.

Secondly, calling `ensureTeamForClerkOrg` after `findCreateUserTeam` indicates that the latter can't be relied upon to do its work correctly. If so, fixing `findCreateUserTeam` would be the solution.

Here's the commit:
https://github.com/onmetal-dev/metal/pull/56/commits/fcd3d0bc8704f7945ef107b9581869d8727c4412

Reply:
I don't agree, it's still very confusing for these two cases to be mashed together in findCreateUserTeam. If they share some logic then factor that logic out separately? If there's twice repeated logic it's not a big deal

Reply:
It also doesn't imply the latter is not working properly--they are doing two completely different things. The active org is a newly created org, whereas the personal team is a team we create to make sure the user is at least part of one team

Reply:
> similarly don't like the extra statefulness of the userClerkOrganizationId param here...
> 
> My understanding of the previous version of this function was that it roughly did:
> 
>     1. try to find a Team that the user created and is part of in the DB. If found, we're done
> 
>     2. create a new org in Clerk and a new Team in our db
> 
> 
> Now there's a new userClerkOrganizationId param that is unclear to me what it means / how it's useful in the context of this function's original purpose of creating a personal team for the user.
> 
> I think it'd be clearer to separate this out entirely and maybe create a separate function call that checks that the active clerk org has a corresponding Team in our db, e.g.
> 
> ```ts
>   const userPersonalTeam = await findCreateUserTeam({
>     teamName: `${user.firstName}'s Projects`,
>     userId: user.id,
>     userClerkId: clerkUser.id,
>   });
> 
>   // new
>   const { orgId } = auth();
>   if (orgId) {
>     await ensureTeamForClerkOrg({ clerkOrgId: orgId })
>   }
> ```

I've reverted my changes to `findCreateUserTeam` and called `ensureTeamForClerkOrg` separately as you requested.